_posts/2022-06-13-mfa-phase-1-rollout.md

```diff
@@ -1,16 +1,18 @@
 ---
 title: Making popular Ruby packages more secure
 layout: post
-author: 
-author_email: 
+author: Betty Li
+author_email: betty.li@shopify.com
 ---
 
-Attacks on the software supply chain are increasing. Our community has not gone unscathed. RubyGems has been affected by supply chain attacks in the past. It’s important for us to mitigate these risks as much as possible. Recommending stronger security practices like enabling multi-factor authentication (MFA) on high-impact packages will be a first step in improving security of the RubyGems ecosystem.
+Attacks on the software supply chain are increasing. Our community has not gone unscathed. RubyGems has been affected by supply chain attacks in the past. It’s important for us to mitigate these risks as much as possible. Recommending stronger security practices like enabling multi-factor authentication (MFA) on popular packages is a first step towards improving security of the RubyGems ecosystem.
 
-Account takeovers are the second most common attack on software supply chains ([source](https://arxiv.org/abs/2002.01139)). The countermeasure against this type of attack is simple: enabling MFA. Doing so will help reduce the chances of a successful attack by 99.9 percent ([source](https://www.microsoft.com/security/blog/2019/08/20/one-simple-action-you-can-take-to-prevent-99-9-percent-of-account-attacks/)).
+Account takeovers are the [second most common](https://arxiv.org/abs/2002.01139) attack on software supply chains. The countermeasure against this type of attack is simple: enabling MFA. Doing so will help [reduce the chances](https://www.microsoft.com/security/blog/2019/08/20/one-simple-action-you-can-take-to-prevent-99-9-percent-of-account-attacks/) of a successful attack by 99.9 percent.
 
-Starting today, top-100 RubyGems package maintainers will begin to see reminders on the RubyGems command-line tool and website if MFA is not enabled on their accounts. Anybody who maintains a gem with more than 165 million downloads will see this recommendation.
+As proposed in the [MFA rollout RFC](https://github.com/rubygems/rfcs/blob/master/text/0007-mfa-rollout.md), RubyGems is beginning to move towards greater adoption of multi-factor authentication for gem maintainers.
 
-Although this is currently just a recommendation, we will enforce MFA on high-impact gems in the future. This policy will bring us in line with other package ecosystems (e.g. [npm](https://github.blog/2022-02-01-top-100-npm-package-maintainers-require-2fa-additional-security/)). 
+Starting today (June 13, 2022), the maintainers of at least the top 100 RubyGems packages will begin to see reminders on the RubyGems command-line tool and website if MFA is not enabled on their accounts. Anybody who maintains a gem with more than 165 million downloads will see this recommendation.
```

Comment:
The thought around `reminders` as the word-choice is because this paragraph is speaking to the changes made for the Recommendation Phase, so the intention was to use softer language.

But! We have been referring to them as `warnings` and you raise an excellent point that the messages are prefixed with it too 😅 . I dig it! Will make the change. 

Also, I good point re: `at least`. Will fix as well.

Reply:
I believe the 165+ mill threshold now exceeds the top 100 packages now (but I guess people don't need to know that). The `at least` sounds kinda wonky to me though. Maybe something like `the maintainers representing the top 100 RubyGems packages` would work?

Reply:
Re: `at least top 100 gems`, Aditya raised [the following point](https://github.com/rubygems/rubygems.github.io/pull/110#discussion_r867525141):

> top-100 and 165 mil+ downloads may not always be accurate. lets talk about download count first and then clarify that we are hoping to target at least top 100 gems by downloads.

I hear where he's coming from. I think keeping `at least` makes sense. What do you reckon @aellispierce?

Reply:
@jenshenny Thank you for the feedback 🙏 . Seeing [your note](https://github.com/Shopify/rubygems.github.io/pull/1#discussion_r888359558) about `at least` too, I'm pulling you into this thread. 

Given how we've actually [implemented](https://github.com/rubygems/rubygems.org/blob/dd407fcfbc03d482e323e2c3680d4a17df2b9689/app/models/user.rb#L241) which users will be classified under "mfa recommended", we are in fact targeting more than just the top 100 gem. 

The truth is we'll be affecting any gem with more than 165 mil+ downloads. I think that's where the spirit of `at least` comes from. 

Unless! Do are we proposing we solely scope it to the top 100? If so, we'll need to revise our implementation so that we check for 165 mil+ downloads **_and_** gem is one of the top 100. 

Thoughts?

Reply:
yep same! +1 with @aellispierce, I'm 👍  with `at least`